ruby/red-arrow/ext/arrow/raw-records.cpp

```diff
@@ -144,6 +144,122 @@ namespace red_arrow {
       // The number of columns.
       const int n_columns_;
     };
+
+    class RawRecordsProducer : private Converter, public arrow::ArrayVisitor {
+    public:
+      explicit RawRecordsProducer(int n_columns)
```

Comment:
Can we remove `n_columns` here and define `n_columns` in each `produce()`?

```cpp
 void produce(const arrow::RecordBatch& record_batch) {
   auto n_columns = record_batch->num_columns();
   // ...
 }
 
 void produce(const arrow::Table& table) {
   auto n_columns = table->num_columns();
   // ...
 }

Reply:
fix: [b0731a7](https://github.com/apache/arrow/pull/37137/commits/b0731a78e6ef5e30d6451db244ffd7f6f7b45a5d)